spec/models/teacher_spec.rb

```diff
@@ -546,6 +546,27 @@
         expect(Teacher.active_in_trs.to_sql).to end_with(expected_clause)
       end
     end
+
+    describe ".mentor_declarations" do
```

Comment:
Fine with this how you have it, but I think it would be safe enough to test this with shoulda matchers `have_many(:mentor_declarations).through(...)`